src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.controller;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Servlet class responsible for the profile page. */
+public class ProfileServlet extends HttpServlet {
+
+  /** Store class that gives access to Messages. */
+  private MessageStore messageStore;
+
+  /** Store class that gives access to Users. */
+  private UserStore userStore;
+
+  /** Tell whether the profile page is on the current logged in User. */
+  private boolean current;
+
+  /** Set up state for handling chat requests. */
```

Comment:
Also not sure if this is necessary - does this servlet handle chat requests?

Reply:
The purpose of the "current" variable is because the profile page should show something different depending on if the page you're looking at is your page or another person's. So, I made this variable for Isaac to use on the front-end to differentiate. If it's your profile, you should be able to edit the description. Otherwise, it should just display the description. And to answer your question, no it doesn't handle chat requests. It only lists messages by that user whose page you're on.